main.tf

```diff
@@ -250,6 +250,135 @@ resource "aws_s3_bucket_notification" "main" {
   }
 }
 
+resource "aws_s3_bucket_replication_configuration" "main" {
+  count = var.replication_configuration != null ? 1 : 0
+
+  depends_on = [aws_s3_bucket_versioning.main]
```

Comment:
aws_s3_bucket_versioning is conditional, don't you need to reference specific index? i.e. `aws_s3_bucket_versioning.main[0]`